crates/iceberg-catalog/src/api/management/mod.rs

```diff
@@ -37,6 +35,109 @@ pub mod v1 {
+                         AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                         Extension(metadata): Extension<RequestMetadata>| {
+                            Self::list_warehouses(project_id.into(), request, api_context, metadata)
+                        },
+                    )
+                )
+                // Delete warehouse - must be empty
+                .route("/warehouse/:warehouse_id", 
+                get(
+                    |Path(warehouse_id): Path<uuid::Uuid>,
+                     AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                     Extension(metadata): Extension<RequestMetadata>| {
+                        Self::get_warehouse(warehouse_id.into(), api_context, metadata)
+                    },
+                )
+                .delete(
+                    |Path(warehouse_id): Path<uuid::Uuid>,
+                     AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                     Extension(metadata): Extension<RequestMetadata>| {
+                        Self::delete_warehouse(warehouse_id.into(), api_context, metadata)
+                    },
+                ))
+                // Rename warehouse
+                .route(
+                    "/warehouse/:warehouse_id/rename",
+                    post(
+                        |
+                        Path(warehouse_id): Path<uuid::Uuid>,
+                        AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                         Extension(metadata): Extension<RequestMetadata>,
+                         Json(request): Json<warehouse::RenameWarehouseRequest>| {
+                            Self::rename_warehouse(warehouse_id.into(), request, api_context, metadata)
+                        },
+                    ),
+                )
+                // Deactivate warehouse
+                .route(
+                    "/warehouse/:warehouse_id/deactivate",
+                    post(
+                        |
+                        Path(warehouse_id): Path<uuid::Uuid>,
+                        AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                         Extension(metadata): Extension<RequestMetadata>| {
+                            Self::deactivate_warehouse(warehouse_id.into(), api_context, metadata)
+                        },
+                    ),
+                )
+                .route(
+                    "/warehouse/:warehouse_id/activate",
+                    post(
+                        |
+                        Path(warehouse_id): Path<uuid::Uuid>,
+                        AxumState(api_context): AxumState<ApiContext<State<A, C, S>>>,
+                         Extension(metadata): Extension<RequestMetadata>| {
+                            Self::activate_warehouse(warehouse_id.into(), api_context, metadata)
+                        },
+                    ),
```

Comment:
Is there something inherently different between deactivate & activate? Could make this a single `POST /warehouse/:id/status`, saves a few lines & makes for a smaller API surface

Reply:
There is - deactivating is something a user might want to do. It happens when someone presses the "delete" button in the UI.
Reactivation is more for recovery. Also the processes might become quite different in the future - that's why I prefer to have two endpoints.